tests/test_generators.py

```diff
@@ -361,6 +359,228 @@ def test_state_serialization(self):
+                number_of_vehicles=num_vehicles,
+                vehicle_capacity=0,
+                time_distances=time_distances,
+                time_window_open=time_window_open,
+                time_window_close=time_window_close,
+                service_time=service_time)
+
+        # Not enough total capacity for total demand
+        with self.assertRaises(ValueError):
+            dwave.optimization.generators.capacitated_vehicle_routing_with_time_windows(
+                demand=[30, 30, 31],
+                number_of_vehicles=3,
+                vehicle_capacity=30,
+                time_distances=time_distances,
+                time_window_open=time_window_open,
+                time_window_close=time_window_close,
+                service_time=service_time)
+
+        # Non-zero depot demand
+        with self.assertRaises(ValueError):
+            dwave.optimization.generators.capacitated_vehicle_routing_with_time_windows(
+                demand=demand,
+                number_of_vehicles=num_vehicles,
+                vehicle_capacity=capacity,
+                time_distances=time_distances,
+                time_window_open=time_window_open,
+                time_window_close=time_window_close,
+                service_time=service_time)
+
+        # Demand vector has zero for depot and customer
+        with self.assertRaises(ValueError):
+            dwave.optimization.generators.capacitated_vehicle_routing_with_time_windows(
+                demand=[0, 44.6, 0],
+                number_of_vehicles=num_vehicles,
+                vehicle_capacity=capacity,
+                time_distances=time_distances,
+                time_window_open=time_window_open,
+                time_window_close=time_window_close,
+                service_time=service_time)
+
+        # Demand vector has zero for customer
+        with self.assertRaises(ValueError):
+            dwave.optimization.generators.capacitated_vehicle_routing_with_time_windows(
+                demand=[22, 44.6, 0],
+                number_of_vehicles=num_vehicles,
+                vehicle_capacity=capacity,
+                time_distances=time_distances,
+                time_window_open=time_window_open,
+                time_window_close=time_window_close,
+                service_time=service_time)
+
+    def test_basics(self):
+        num_vehicles = 2
+
+        model = dwave.optimization.generators.capacitated_vehicle_routing_with_time_windows(
+            demand=[0, 5, 5],
+            number_of_vehicles=num_vehicles,
+            vehicle_capacity=10,
+            time_distances=[[1, 6, 3], [2, 2, 8], [4, 6, 7]],
+            time_window_open=[0, 0, 0],
+            time_window_close=[20, 20, 20],
+            service_time=[0, 0, 0])
+
+        self.assertEqual(model.num_decisions(), 1)
+        self.assertEqual(model.num_constraints(), 12)
+        self.assertEqual(model.num_nodes(), 177)
+        self.assertEqual(model.num_edges(), 286)
```

Comment:
I am not sure these tests help much. At the least we should change to `self.assertLessEqual(...)` to support future improvements. But IMO we should drop them.

Reply:
Sounds reasonable. I'm going to drop those.